Preparing to release version 0.8.4

DanHickstein opened the pull request:
Okay, there are no current PRs, so it's as good a time as any to release our long-overdue v0.8.4, right?

It's been so long, that I almost completely forget how to release a new version. As far as I recall, I just need to make a PR to increment the version number and update the changelog. I did these things, and I updated CONTRIBUTING.rst to list these steps. Finally, I will need to make a new release on GitHub and Travis should ship it off to PyPi automatically.

Let me know if there are more things that I should do in this PR to prepare for the new release.

Thanks!

MikhailRyazanov commented:
Looks good! Although the release date is optimistic. :–)

Let's also draft the release notes because it's important to mention there the incompatible changes:
1. The `center` → `origin`/`method` change (this should still work, but with warnings).
2. That (row, column) is now used everywhere (this can break some old code).

stggh reviewed (approved):
Looks fine. Nothing too strenuous for my brain.

DanHickstein commented:
Okay, here is what I have for release notes:

PyAbel 0.8.4 provides a number of valuable updates. It is the first version to incorporate the rBasex method, a powerful method for transforming images obtained from velocity-map-imaging (VMI) experiments. The conventions for defining the origin of the image in (row, column) format has been universally applied to all functions in PyAbel, so users should check for compatibility with this new version. Additionally, the "center" keyword has been deprecated in favor of "origin". 

Here is a full list of changes in v0.8.4:
* Added odd angular orders to tools.vmi.Distributions (PR #266).
* Important! Some "center" functions/parameters are renamed to "origin" or
  "method"; using old names still works but will print deprecation warnings,
  please update your code accordingly. Image origin is now always in the
  (row, column) format for consistency within PyAbel and with NumPy/SciPy; this
  can break some code, so please check carefully and update it if necessary.
  See PR #267.
* Fixed the GUI examples (example_GUI.py and example_simple_GUI.py)
  so that they work with the lastest versions of tk (PR #269).
* New method rBasex for velocity-map images, based on pBasex and the work of
  Mikhail Ryazanov (PR #270).
* Added "orders", "sinpowers" and "valid" to tools.vmi.Distributions results,
  reordered cossin() powers for consistency (PR #270).
* Improved tools.vmi.Distributions performance on Windows (PR #270).
* More corrections to the GUI example: working without the "bases" directory,
  loading "from transform", interface enhancements (PR #277).
* Improved documentation (PR #283, PR #288).
* Correctly use quadrants in abel.Transform (PR #287).
* Circularization now uses periodic splines (to avoid discontinuity), with
  smoothing determined by the RMS tolerance instead of the nonintuitive
  "smooth" parameter (PR #293).
* Corrected and improved tools.center (PR #302).
* Moved numpy import to try block in setup.py. This allows pip to install PyAbel
  in situations where numpy is not already installed. (PR #310)
